packages/graphql/src/translate/create-projection-and-params.ts

```diff
@@ -594,10 +594,48 @@ function createProjectionAndParams({
         return res;
     }
 
+    const existingProjection = { ...resolveTree.fieldsByTypeName[node.name] };
+
+    // if we have a global node, and the selection set includes the id field
+    if (node.isGlobalNode() && existingProjection.id) {
+        const globalIdField = node.getGlobalIdField();
+        // check to see if the idField has been projected, and include it if it hasn't
+        if (!Object.values(existingProjection).find((field) => field.name === globalIdField)) {
+            existingProjection[globalIdField] = {
+                alias: globalIdField,
+                args: {},
+                fieldsByTypeName: {},
+                name: globalIdField,
+            };
+        }
+        if (globalIdField !== "id" && !!existingProjection.id) {
+            delete existingProjection.id;
+        }
```

Comment:
**question**: I don't completely understand the logic behind this bit, but why do we need to remove the id field in this case?

Reply:
Let's take two examples:

```graphql
type ExampleOne @node(global: true) {
  title: String! @unique
}

type ExampleTwo @node(global: true) {
  dbId: ID! @id @alias(property: "id")
}
```

For ExampleOne, if you query for the `id` -- i.e. the globalId -- then we need to add `title` to the selectionSet. But we *also* have to remove `id` from the selectionSet otherwise the cypher will attempt to project a field that doesn't exist and we get an error.

For ExampleTwo, if you query for the `id`, we don't need to do anything except let the `id` selection through since it maps to the underlying db field.

So this second conditional checks to see if the globalIdField is "id." If it isn't and the `id` selection is still in the projection, we need to delete it. 

I'm going to refactor this so it's clearer, but that's the explanation!

Reply:
mmm, I see. forgive me if this is a stupid question, but what would happen in a situation like this?:
```graphql
type ExampleOne @node(global: true) {
  title: String! @unique
  dbId: ID! @id @alias(property: "id")
}
```

We would add the global Id (I assume it would be title in this case), but will remove id from the projection

Reply:
This is an interesting question because it gets to whether we should have the user define the globalIdField or we should infer it.

The current inference strategy looks for an `@id` directive first and `@unique` directives second, so in this case, the globalIdField would be "id" and the id would not get deleted.